pareto/utilities/process_data.py

```diff
@@ -0,0 +1,838 @@
+    )
+
+    strategic_model.p_gamma_TimePeriodDemand = Param(
+        strategic_model.s_T,
+        units=strategic_model.model_units["volume"],
+        doc="Total water demand over the planning horizon [volume]",
+        mutable=True,  # Mutable Param - can be changed in sensitivity analysis without rebuilding the entire model
+    )
+
+    strategic_model.p_sigma_WaterAvailable = Param(
+        strategic_model.s_T,
+        units=strategic_model.model_units["volume"],
+        doc="Total PW and external water available [volume]",
+        mutable=True,  # Mutable Param - can be changed in sensitivity analysis without rebuilding the entire model
+    )
+
+    for t in strategic_model.s_T:
+        strategic_model.p_beta_TotalPW[t] = sum(
+            strategic_model.p_beta_Production[p, t]
+            + strategic_model.p_beta_Flowback[p, t]
+            for p in strategic_model.s_P
+        )
+
+        strategic_model.p_sigma_MaxPWCapacity[t] = (
+            sum(
+                strategic_model.p_gamma_Completions[p, t]
+                for p in strategic_model.s_P  # Completions Demand
+            )
+            + sum(
+                strategic_model.p_omega_EvaporationRate
+                for s in strategic_model.s_S  # Storage Evaporation
+            )
+            + sum(
+                strategic_model.p_sigma_BeneficialReuse[o, t]
+                if strategic_model.p_sigma_BeneficialReuse[o, t].value >= 0
+                else strategic_model.p_M_Flow
+                for o in strategic_model.s_O  # Beneficial Reuse (if user does not specific value, p_sigma_BeneficialReuse is -1)
+                # In this case, Beneficial Reuse at this site has no limit (big M parameter)
+            )
+            + sum(
+                strategic_model.p_sigma_Storage[s]
+                + _get_max_value_for_parameter(strategic_model.p_delta_Storage)
+                for s in strategic_model.s_S  # Storage
+            )
+            + sum(
+                (
+                    strategic_model.p_sigma_Disposal[k]
+                    + _get_max_value_for_parameter(strategic_model.p_delta_Disposal)
+                )
+                * strategic_model.p_epsilon_DisposalOperatingCapacity[k, t]
+                for k in strategic_model.s_K  # (Initial Disposal + max disposal) * operating capacity
+            )
+            + sum(
+                max(
+                    [
+                        (
+                            strategic_model.p_sigma_Treatment[r, wt].value
+                            + max(
+                                [
+                                    strategic_model.p_delta_Treatment[wt, j].value
+                                    for j in strategic_model.s_J
+                                ]
+                            )
+                        )
+                        * (1 - strategic_model.p_epsilon_Treatment[r, wt].value)
```

Comment:
```suggestion
```
I don't think you need this line. By including it, you are essentially discounting the treatment residual from the total PW capacity of the network.

Reply:
It depends on if we think of treatment centers as an endpoint for water OR a node that has some loss. 
Here I've treated it as a node that has some loss. Looking at feasibility, we then want to only include the residual as the end point for water not the full volume of treated water. 
I think we tend to view treatment sites more as a node with loss as water cannot end at the treatment site. 

Does you agree?